core/src/main/java/io/grpc/ClientCallImpl.java

```diff
@@ -127,49 +144,122 @@ public void start(Listener<RespT> observer, Metadata.Headers headers) {
       headers.put(HttpUtil.USER_AGENT_KEY, userAgent);
     }
 
+    boolean started = startInternal(listener, headers);
+    // Start the deadline timer after stream creation because it will close the stream
+    if (started && deadlineNanoTime != null) {
+      deadlineCancellationFuture = startDeadlineTimer(timeoutMicros);
+    }
+  }
+
+  private boolean startInternal(
+      final ClientStreamListener listener, final Metadata.Headers headers) {
+    ClientTransport transport;
     try {
-      stream = transport.newStream(method, headers, listener);
+      transport = clientTransportProvider.get();
+    } catch (RuntimeException ex) {
+      closeCallPrematurely(listener, Status.fromThrowable(ex));
+      return false;
+    }
+    if (transport == null) {
+      closeCallPrematurely(listener, Status.UNAVAILABLE.withDescription("Channel is shutdown"));
+      return false;
+    }
+
+    try {
+      synchronized (this) {
```

Comment:
stream is access potentially by two threads: the application thread and the scheduledExecutor thread.  Because the presence or absence of the stream variable indicates that a connection was established, I need to lock access to it.